lib/pbio/src/color/util.c

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2022 The Pybricks Authors
+
+#include <pbio/color.h>
+
+int32_t sin_deg_branch0(int32_t x) {
+    return 201 * x - x * x;
+}
+
+// integer sine approximation from degrees to (-10000, 10000)
+int32_t sin_deg(int32_t x) {
+    x = x % 360;
+    if (x < 90) {
+        return sin_deg_branch0(x);
+    }
+    if (x < 180) {
+        return sin_deg_branch0(180 - x);
+    }
+    if (x < 270) {
+        return -sin_deg_branch0(x - 180);
+    }
+    return -sin_deg_branch0(360 - x);
+}
+
+int32_t cos_deg(int32_t x) {
+    return sin_deg(x + 90);
+}
+
+// gets squared cartesian distance between hsv colors mapped into a chroma-lightness-bicone
+int32_t pbio_get_bicone_cost(const pbio_color_hsv_t *hsv_a, const pbio_color_hsv_t *hsv_b) {
```

Comment:
Since this is measuring distance, can we call it `pbio_get_bicone_distance` instead of `pbio_get_bicone_cost`?

Reply:
I probably stuck to cost since this actually returns the squared distance. So it probably should be either `pbio_get_bicone_cost` or `pbio_get_bicone_squared_distance`?

Reply:
The meaning of "cost" in this context isn't intuitive to me, so it would be nice to either explain what it means in the doc comments or use the more explicit name.

Reply:
OK, then I would opt for "squared distance". Cost only makes sense in the context of the color map.